filters.go

```diff
@@ -311,6 +311,38 @@ func (p *Pipe) MatchRegexp(re *regexp.Regexp) *Pipe {
 	})
 }
 
+// ReadInput reads displays the pipe content as a message as a prompt
```

Comment:
One question that occurs with this is how to test it! Hard-wiring `os.Stdin` is not great for several reasons, but this is a larger issue with `script` in general; I'd like the pipe's standard in and standard out to be reassignable. When we have that, this issue will go away.

Reply:
That's a great point. I personally didn't like it either to hard wire `ReadInput` to `os.Stdin` but as you pointed out, solving it seemed a bit out of scope.

My suggestion is creating two `ReadInput` versions, (since Go doesn't allow function overloading or default parameters).

One package-scoped and private 
```go
func (p *Pipe) readInput(defaultValue string, stdin io.Reader) *Pipe
```
And the public one:
```go
func (p *Pipe) ReadInput(defaultValue string) *Pipe {
    return p.readInput(defaultValue, os.Stdin())
}
```

And in the tests, we will have to convince ourselves that testing `readInput` is good enough. And that this compromise will save us the trouble of stubbing `os.Stdin()` in a safe way.

Reply:
I'm using `TestMain` to do this for some tests (see `TestStdin`, for example).

Reply:
I did my best effort following the example of `TestStdin` and keeping the tests thread-safe